src/mad_spark_alt/core/json_utils.py

```
@@ -182,6 +187,219 @@ def validate_json_structure(data: Dict[str, Any], required_keys: list) -> bool:
+    required_keys = ["content", "mutation_type", "reasoning"]
+    if not all(key in mutated_idea for key in required_keys):
+        return False
+
+    for key in required_keys:
+        if not _is_valid_non_empty_string(mutated_idea[key]):
+            return False
+
+    return True
+
+
+def validate_selection_response(data: Dict[str, Any]) -> bool:
```

Ruff (0.12.2)

`validate_selection_response` is too complex (14 > 10)

(C901)